### DISCUSSION TITLE
Should I rebuild the project when important changes are pulled?

### DISCUSSION THREAD

**Discussion by vitorfrois** [courselab/snaskii22#26] [Q&A]
Is it necessary to rebuild the project when critical changes are made and pulled from the main/develop branch?

**Comment by vrrusso** [comment]
This is an interest question, from my understanding ./boostrap.sh build the configuration for your system, and ./configure make the necessary adjustment. I think only make && make install is required.

**Comment by monacofj** [comment]
If you change only the source code (no matter how complex is the change) you may just run `make`.
If you change something in the build scripts (`configure.ac`, `Makefine.am`), then you need to rerun `bootstrap.sh` to recriate `configure`, and then rerun `configure`to recreate the `Makefile`s. 

You may need do modify the build script if you want to apply some new compiler flag, or if you ad new source code.

**Reply by jj-marinho** [reply]
By the way, if anyone is trying to add new scenes, make sure to take a look at scenes/Makefile.am and recompile the code.

I was having problems because new scenes aren't automatically added to the build script, so even if I tried rebuilding they wouldn't work. 

Changing this: https://github.com/courselab/snaskii22/pull/91/files#diff-16f908fd22fb62ea2011f2557b5cd4ddc7385d7d018669abd960174f2a29486a

And then rebuilding worked.
